mmv1/templates/terraform/pre_create/spanner_instance_config.go.erb

```diff
@@ -0,0 +1,9 @@
+baseReplicas, err := getBaseInstanceConfigReplicas(d, config, baseConfigProp, billingProject, userAgent)
+if err != nil {
+  return err
+}
+l := obj["instanceConfig"].(map[string]interface{})
+r := l["replicas"].([]interface{})
+r = append(r, baseReplicas...)
+l["replicas"] = r
+obj["instanceConfig"] = l
```

Comment:
Only reason we want to silently add them because those are not configurable all the replicas of baseConfig need to be present, the only choice customer have is to choose from optional_replicas(another field), since API does not handle this from backend we need to silently add them and pass in API.

https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_create_instance_config.go#L61C7-L63